python/ql/src/semmle/python/essa/SsaCompute.qll

```diff
@@ -482,6 +501,19 @@ private module SsaComputeImpl {
       b = def.(PhiFunction).getBasicBlock() and
       i = -1
     }
+
+    /**
+     * Holds if the value defined at `def` can reach `use`, possibly through phi nodes.
+     */
+    cached
+    predicate aUse(EssaDefinition def, ControlFlowNode use) {
+      firstUse(def, use)
+      or
+      exists(ControlFlowNode firstUse |
+        firstUse(def, firstUse) and
+        adjacentUseUse(firstUse, use)
+      )
```

Comment:
Current definition only allows `def -> firstUse`, and `def -> firstUse -> secondUse`, and not `def -> firstUse -> secondUse -> thirdUse`. That seems like a problem :stuck_out_tongue: 

```suggestion
      exists(ControlFlowNode mid |
        aUse(def, mid) and
        adjacentUseUse(mid, use)
      )
```

Reply:
this was fixed by now I see.